### DISCUSSION TITLE
(postgres) defaultValue is not enforced in database

### DISCUSSION THREAD

**Discussion by HarleySalas** [payloadcms/payload#6048] [Feature Requests & Ideas]
In my scenario, I'm using a `radio` field, with `required: true` and `defaultValue: 'myValue'`. I have these things on globals, so they're at the base level of the table, not in any sort of creatable state. (It's a typical "link" field, as in the website example)

According to discord user `ritsu0455`, he believes that every field's `defaultValue` is not enforced at the schema level, so it could create issues for many scenarios.

If it cannot be enforced at the database level, as I understand that fields are flexible elements, perhaps you could simply handle scenarios where when an existing document has a field with a defaultValue added, it should at least apply that value when being added.

**Comment by jmikrut** [comment]
Hey @HarleySalas — we actually _**do**_ populate default values if an existing document is **_resaved_** after having a new `defaultValue` added. But I suspect your issue is more related to database migrations (i.e. enforcing `required: true` after having null columns present). 

We _**could**_ enforce statically defined default values on the DB-level, but as you said, Payload's `defaultValue` is able to be defined as a function which would not work for DB-level enforcement.

Enforcing statically defined default values on the DB level in Postgres would certainly be a good feature request, and wouldn't be hard to do. But I'm not sure that we will be able to get to it in the immediate future. I will convert this to a Discussion and transfer it over to the Payload core repo so that we can track it accordingly as a feature request. 

How's that?

**Comment by r1tsuu** [comment]
Hi @HarleySalas @jmikrut  this PR adds https://github.com/payloadcms/payload/pull/6983 non-function `defaultValue` DB enforcing

**Comment by DanRibbens** [comment]
I had to do this as a separate PR. https://github.com/payloadcms/payload/pull/7368
